pkg/providers/launchtemplate/suite_test.go

```diff
@@ -548,6 +548,16 @@ var _ = Describe("LaunchTemplates", func() {
 			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
 			ExpectNotScheduled(ctx, env.Client, pod)
 		})
+		It("should pack pods if the sum of pod ephemeral-storage and overhead exceeds node capacity and instance storage is mounted", func() {
+			nodeClass.Spec.InstanceStorePolicy = lo.ToPtr(v1beta1.InstanceStorePolicyRAID0)
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod(coretest.PodOptions{ResourceRequirements: v1.ResourceRequirements{
+				Requests: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceEphemeralStorage: resource.MustParse("19Gi"),
+				}}})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+			ExpectScheduled(ctx, env.Client, pod)
```

Comment:
We should probably also consider checking that we passed the correct userData through when we are using the AL2 ami here

Reply:
Good call, added https://github.com/aws/karpenter-provider-aws/pull/4735/commits/4bf56c65efa9007b3aee0871e6d554e370fa549b